src/core/column-list/column.list.model.d.ts

```diff
@@ -0,0 +1,16 @@
+import {ColumnModel} from "../column-type/column.model";
+
+export declare enum Generation{
```

Comment:
Better make it a type like this:
`type Generation = 'deep' | 'shallow';`
https://basarat.gitbooks.io/typescript/docs/types/literal-types.html